packages/perf/lib/modular/index.js

```diff
@@ -0,0 +1,70 @@
+import { isBoolean } from '@react-native-firebase/app/lib/common';
+import { firebase } from '..';
+
+/**
+ * Returns a Performance instance for the given app.
+ * @param app - FirebaseApp. Optional.
+ * @returns {Performance}
+ */
+export function getPerformance(app) {
+  if (app) {
+    return firebase.app(app.name).perf();
+  }
+
+  return firebase.app().perf();
+}
+
+/**
+ * Returns a Performance instance for the given app.
+ * @param app - FirebaseApp. Required.
+ * @param settings - PerformanceSettings. Set "dataCollectionEnabled" which will enable/disable Performance collection.
+ * @returns {Performance}
+ */
+export async function initializePerformance(app, settings) {
+  const perf = firebase.app(app.name).perf();
+
+  if (settings && isBoolean(settings.dataCollectionEnabled)) {
+    await perf.setPerformanceCollectionEnabled(settings.dataCollectionEnabled);
+  }
+
+  return perf;
+}
+
+/**
+ * Determines whether performance monitoring is enabled or disabled.
+ * @param perf - Performance instance
+ * @returns {Boolean}
+ */
+export function isPerformanceCollectionEnabled(perf) {
+  return perf.isPerformanceCollectionEnabled;
+}
+
+/**
+ * Returns nothing.
+ * @param perf - Performance instance
+ * @param enabled - A Boolean for setting Performance collection
+ * @returns {void}
+ */
+export function setPerformanceCollectionEnabled(perf, enabled) {
+  return perf.setPerformanceCollectionEnabled(enabled);
+}
+
+/**
+ * Returns a Trace instance.
+ * @param perf - Performance instance
+ * @param identifier - A String to identify the Trace instance
+ * @returns {Trace}
+ */
+export function trace(perf, identifier) {
+  return perf.newTrace(identifier);
+}
+
+/**
+ * Returns a HttpMetric instance.
+ * @param perf - Performance instance
+ * @param identifier - A String to identify the HttpMetric instance
+ * @returns {HttpMetric}
+ */
+export function httpMetric(perf, identifier, httpMethod) {
```

Comment:
and this can be removed as well. It isn't a web API

Reply:
I think this is one where we need to leave it. Native is a superset of web in general and this is a function that is on android and ios and is useful. It does not represent an incompatibility because if you were on web you don't care if there is an extra feature, but if you need it on native you will want it exposed.

Then for a web implementation (in future) we would stub it instead of calling in to JS API